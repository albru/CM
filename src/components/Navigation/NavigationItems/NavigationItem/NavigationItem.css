.NavigationItem {
    padding-bottom: 10px;
    width: 100%;
    height: 50px;
    display: block;
}

.NavigationItem a {
    display: inline-block;
    text-decoration: none;
    cursor: pointer;
    text-transform: uppercase;
    color: var(--color-gray);
    text-transform: initial;
    font-weight: bold;
    width: 100%;
    height: 100%;
}

@media (min-width: 500px) {
    .NavigationItem {
        display: flex;
        align-items:center;
        justify-content: center;
        margin-bottom: 0;
        height: 100%;
        width: inherit;
        padding: 0;
    }
    .NavigationItem:not(:last-child) {
        margin-right: 15px;
    }

    .NavigationItem a{
        transition: all .3s;
        border-bottom: 1px solid transparent;
        width: auto;
        height: auto;
    }

    .NavigationItem a:hover {
        color: var(--color-main);
        border-bottom: 1px solid var(--color-main);
    }
    .NavigationItemActive {
        color: var(--color-main)!important;
        border-bottom: 1px solid var(--color-main)!important;
    }
}
