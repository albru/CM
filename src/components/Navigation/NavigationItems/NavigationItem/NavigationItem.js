import React from 'react';
import { NavLink } from 'react-router-dom';

import classes from './NavigationItem.css';

const NavigationItem = props => {
    const navigationLinkData = [
        {name: "Наружная Реклама", route: "/1", exact: null},
        {name: "Рекламные услуги", route: "/2", exact: null},
        {name: "Портфолио", route: "/3", exact: null},
        {name: "Контакты", route: "/4", exact: null},
    ];
    return navigationLinkData.map(item => {
        return ( 
            <li key={item.route}
                className={classes.NavigationItem}>
                <NavLink 
                    to={item.route}
                    exact={item.exact}>{item.name}</NavLink>
            </li>
            )
    })
};

export default NavigationItem;