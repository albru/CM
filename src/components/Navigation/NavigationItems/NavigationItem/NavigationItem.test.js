import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import NavigationItem from './NavigationItem';

configure({ adapter: new Adapter() })

describe('NavigationItem', () => {
    it('should render correctly', () => {
        const output = shallow(
            <NavigationItem name="mockName" route="mockRoute" />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});