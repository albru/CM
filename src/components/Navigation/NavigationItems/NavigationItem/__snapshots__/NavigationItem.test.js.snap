// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem should render correctly 1`] = `
<li>
  <NavLink
    to="mockRoute"
  >
    mockName
  </NavLink>
</li>
`;
